### ISSUE TITLE
E3037 not triggered for hard coded key values in AWS::ECS::TaskDefinition

### ISSUE THREAD

**Issue by bgardner-noggin** [aws-cloudformation/cfn-lint#2930]
### CloudFormation Lint Version

0.83.0

### What operating system are you using?

Fedora

### Describe the bug

E3037 does not get triggered for AWS::ECS::TaskDefinition, ContainerDefinitions -> Environment for hard coded keys

In the attached zip are two templates. Both of them should trigger the E3037 error however the one with environment keys

`Environment:
  - Name: DUPE_KEY
    Value: 'firstval'
  - Name: DUPE_KEY
    Value: 'secondval'
`
does not error

[e3037-recreate.zip](https://github.com/aws-cloudformation/cfn-lint/files/13171806/e3037-recreate.zip)

### Expected behavior

E3037 should be triggered for hard coded keys

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: "Dupe env var recreate case"

Parameters:
    RoleARN:
        Type: String

Resources:

    TaskDefinitionFG:
        Type: AWS::ECS::TaskDefinition
        Properties:
            NetworkMode: awsvpc
            Family: "myservice"
            RequiresCompatibilities: [ FARGATE ]
            RuntimePlatform:
                CpuArchitecture: X86_64
                OperatingSystemFamily: LINUX
            TaskRoleArn: !Ref RoleARN
            ExecutionRoleArn: !Ref RoleARN
            Cpu: 512
            Memory: 1024
            ContainerDefinitions:
              - Name: "mycontainer"
                User: "user"
                Cpu: 512
                Memory: 1024
                Essential: true
                StopTimeout: 40
                Image: !Sub "${AWS::AccountId}.dkr.ecr.${AWS::Region}.amazonaws.com/repo:latest"
                Command:
                  - 'run'
                LinuxParameters:
                    InitProcessEnabled: true
                Environment:
                  - Name: DUPE_KEY
                    Value: 'firstval'
                  - Name: DUPE_KEY
                    Value: 'secondval'
```

**Comment by kddejong** [comment]
This is an interesting one.  We will have to create a new rule for this.  So E3037 will flag on all elements not just one key.  

So this should have flagged.  What we are really looking at here is that 
```yaml
Environment:
- Name: DUPE_KEY
   Value: 'firstval'
- Name: DUPE_KEY
   Value: 'firstval'
```

The resulting task definition drops the secondary Name(s) which isn't an ideal experience.  Going to ask some internal people on how we should handle this.
```json
"environment": [
  {
    "name": "DUPE_KEY",
    "value": "firstval"
  }
]
```

**Comment by bgardner-noggin** [comment]
The AWS result is worse than that. Out of 7 stacks we had installed through different accounts / regions, one of them, and only after one update, used the first value.

**Comment by kddejong** [comment]
I've included a fix for this in v1.  I still have some work to do for adding this to Tags.
